### PR TITLE
Update Alert class to handle nrql validation

### PR DESCRIPTION
**Pull request by josephgregoryii**
# Summary

This PR stems from handling alert policy error while installing alerts in our UI.

We found that alert conditions that contain a nrql query with `TIMESERIES` fail an installation after it 
hits our database.

This is a supplementary PR #2190 so we can have a more automated process when verifying alert conditions in PRs.

### Testing

Our workflows that validate quickstarts will fail during checks.

Testing PR: https://github.com/newrelic/newrelic-quickstarts/pull/2197
- This PR should fail

Link to validation failure + logs: https://github.com/newrelic/newrelic-quickstarts/actions/runs/7468258398/job/20323336084

> Note: as of right now, we have 3 other alerts containing invalid queries, so it is a little noisy in the logs

### CODE REVIEW

File: utils/lib/__tests__/Quickstart.test.js [hunk 1]
```diff
@@ -100,6 +100,37 @@ describe('Quickstart', () => {
...
+          return component;
+        }
+      }).filter(Boolean);
+
+      qs.validate();
+
+      expect(invalidComponents).toHaveLength(1)
+
+      const invalidComponent = invalidComponents.pop();
+      expect(invalidComponent.identifier).toBe('mock-alert-policy-3')
```
Comment (zstix) [inline utils/lib/__tests__/Quickstart.test.js:129]: Okay so this is me being totally unreasonably critical and I well welcome you telling me this is too nitpicky, but I'd advocate for us avoiding `pop` and using `[0]` (optionally chained, if needed) inline instead. Reason being is that it alters the shape of `invalidComponents` variable which could be an issue if we decide to make further assertions on that variable.

**Review by zstix — APPROVED** [review]
A bunch of nitpicky style-based things. Nothing I would block the PR on, but if you _do_ decide to incorporate those fixes let me know and I'll re-review.

File: utils/lib/__tests__/Quickstart.test.js [hunk 2]
```diff
@@ -100,6 +100,37 @@ describe('Quickstart', () => {
...
+
+      expect(components).toBeDefined();
+      expect(qs.isValid).toBe(true);
+
+      // all components are invalid
+      const invalidComponents = components.map((component) => {
+        if (!component.isValid) {
+          return component;
+        }
+      }).filter(Boolean);
```
[the remark this answers is not in the document]
Comment (josephgregoryii) [reply utils/lib/__tests__/Quickstart.test.js:122]: Here i am, thinking i was clever 😂  you're totally correct on this

File: utils/lib/__tests__/Quickstart.test.js (hunk 1 above)
Comment (josephgregoryii) [reply utils/lib/__tests__/Quickstart.test.js:129]: Don't thank thats unreasonable at all! I'll update that

File: utils/lib/Alert.ts [hunk 3]
```diff
@@ -72,6 +83,11 @@ export type SubmitSetRequiredDataSourcesMutationResult =
   | { errors: ErrorOrNerdGraphError[] };
 
 class Alert extends Component<QuickstartConfigAlert[], QuickstartAlertInput[]> {
+  constructor(identifier: string, basePath = path.join(__dirname, '..', '..')) {
+    super(identifier, basePath);
```
[the remark this answers is not in the document]
Comment (josephgregoryii) [reply utils/lib/Alert.ts:87]: Ah, i had a a feeling this felt a bit redundet, thanks!

**Review by mickeyryan42 — APPROVED** [review]
This lgtm!

**Pull request merged**